app/controllers/api/v0/api_controller.rb

```diff
@@ -81,13 +81,12 @@ def authenticate_with_api_key_or_current_user!
         error_unauthorized unless user
       end
 
-      def add_missing_api_key_warning_header
+      def add_deprecation_warning_header
         return if headers["Accept"].present? &&
-          headers["Accept"].include?("application/vnd.forem.api-v#{@version}+json") &&
-          headers["api-key"].present?
+          headers["Accept"].include?("application/vnd.forem.api-v#{@version}+json")
 
         # rubocop:disable Layout/LineLength
-        response.headers["Warning"] = "299 - This endpoint will require the `api-key` header and the `Accept` header to be set to `application/vnd.forem.api-v1+json` in future."
+        response.headers["Warning"] = "299 - This endpoint is part of the V0 (beta) API. To start using the V1 endpoints add the `Accept` header and set it to `application/vnd.forem.api-v1+json`. Visit https://developers.forem.com/api for more information."
```

Comment:
Since we've rolled back the CORS + api key requirement on some of the endpoints, a problem arises with the logic when displaying the warning header in the API V0 endpoint. This is the easiest update I came up with (also taking into account changes added by #18246). Very open to other's thoughts/alternatives on this though